pilot/pkg/xds/ads.go

```diff
@@ -103,26 +95,48 @@ type Connection struct {
 	stop chan struct{}
 
 	// reqChan is used to receive discovery requests for this connection.
-	reqChan      chan *discovery.DiscoveryRequest
-	deltaReqChan chan *discovery.DeltaDiscoveryRequest
+	reqChan chan *discovery.DiscoveryRequest
 
 	// errorChan is used to process error during discovery request processing.
 	errorChan chan error
 }
 
-func (conn *Connection) ID() string {
+type Connection struct {
+	BaseConnection
```

Comment:
why do you define a BaseConnection

Reply:
`model.Proxy` is highly coupled to pilot. Base connection can be shared with proxy agent, so we have to split it into two.

Reply:
Then eventually it  should be moved topkg/xds, right?

And `BaseConnection` is very general name, could be more concrete, like adding xds prefix

Reply:
Yeah, that's the idea. Not moving in the same PR since it's easier to see the changes.

Reply:
I can change it to just `Connection` when I move it. `XdsConnection` would be repetitive in `xds` package.